### ISSUE TITLE
net: TestLookupDotsWithLocalSource fails

### ISSUE THREAD

**Issue by darkfeline** [golang/go#29712]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.11.4 linux/amd64
</pre>

### Does this issue reproduce with the latest release?

Yes, 1.11.4 is the latest (not counting betas?).

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/ionasal/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/ionasal/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/lib/go"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build457575040=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?

```
mkdir -p /tmp/tmp3
cd /tmp/tmp3
go mod init example.com/test
cat >main.go <<EOF
package foo

import (
	_ "net"
)
EOF
go test all
```

### What did you expect to see?

All tests pass

### What did you see instead?

Test failures:

```
--- FAIL: TestLookupDotsWithLocalSource (0.00s)
    lookup_test.go:612: netgo: got localhost.; want localhost
FAIL
FAIL	net	29.770s
```

This appears to be a regression of or related to https://github.com/golang/go/issues/15881

**Comment by bcmills** [comment]
CC @mikioh @bradfitz @ianlancetaylor for `net`

**Comment by bcmills** [comment]
This test appears to be a regression test for #13564, and the test was refactored to be more permissive in #14050.

**Comment by darkfeline** [comment]
This is on Arch Linux.

```
$ uname -v
#1 SMP PREEMPT Wed Jan 16 22:38:58 UTC 2019
$ uname -a
Linux myhost 4.20.3-arch1-1-ARCH #1 SMP PREEMPT Wed Jan 16 22:38:58 UTC 2019 x86_64 GNU/Linux
```

(no lsb-release provided by default)

**Comment by darkfeline** [comment]
I fixed this by adding `127.0.0.1 localhost` to `/etc/hosts`.
